cmd/av/stack_branch.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"emperror.dev/errors"
+	"github.com/aviator-co/av/internal/git"
+	"github.com/aviator-co/av/internal/meta"
+	"github.com/aviator-co/av/internal/utils/cleanup"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var stackBranchFlags struct {
+	// The parent branch to base the new branch off.
+	// By default, this is the current branch.
+	Parent string
+}
+var stackBranchCmd = &cobra.Command{
+	Use:   "branch [flags] <branch-name>",
+	Short: "create a new stacked branch",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			_ = cmd.Usage()
+			return errors.New("exactly one branch name is required")
+		}
+
+		var cu cleanup.Cleanup
+		defer cu.Cleanup()
+
+		repo, err := getRepo()
+		if err != nil {
+			return err
+		}
+
+		name := args[0]
+
+		// validate operation preconditions
+		if _, err := repo.RevParse(&git.RevParse{Rev: name}); err == nil {
+			return errors.Errorf("branch %q already exists", name)
+		}
+
+		// Determine important contextual information from Git
+		defaultBranch, err := repo.DefaultBranch()
+		if err != nil {
+			return errors.WrapIf(err, "failed to determine repository default branch")
+		}
+
+		// Determine the parent branch and make sure it's checked out
+		var parentBranch string
+		if stackBranchFlags.Parent != "" {
+			parentBranch = stackBranchFlags.Parent
+			origBranch, err := repo.CheckoutBranch(&git.CheckoutBranch{Name: parentBranch})
+			if err != nil {
+				return errors.WrapIf(err, "failed to checkout parent branch")
+			}
+			cu.Add(func() {
+				if _, err := repo.CheckoutBranch(&git.CheckoutBranch{Name: origBranch}); err != nil {
+					logrus.WithError(err).Warn("cleanup error: failed to return to original branch")
+				}
+			})
+		} else {
+			var err error
+			parentBranch, err = repo.CurrentBranchName()
+			if err != nil {
+				return errors.WrapIff(err, "failed to get current branch name")
+			}
+		}
+
+		// Special case: branching from the repository default branch
+		if parentBranch == defaultBranch {
```

Comment:
can you document in code this behavior - we are intentionally setting the parent Branch for root branch as empty, so that we can uniquely identify a tree.